### ISSUE TITLE
Consider using direct dependency version strings

### ISSUE THREAD

**Issue by oliviercailloux** [ngeor/archetype-quickstart-jdk8#17]
Thanks for this archetype, it’s much needed as a replacement of the default one, by now quite outdated. (I have been working on a new archetype [myself](https://github.com/oliviercailloux/java-archetype).)

I see that this archetype generates a [pom](https://github.com/ngeor/archetype-quickstart-jdk8/blob/master/src/main/resources/archetype-resources/pom.xml) which uses [indirect version strings](https://stackoverflow.com/q/23733849) for its dependencies. I wonder if this is something you might consider changing, or perhaps explain the rationale behind that choice?

I admit it is a common choice, but I hardly see its benefits. As a drawback, it makes the pom slightly longer and more complex. I feel this is unjustified.

I admit there are some benefits in special cases (such as when multiple artifacts have the same version), but I am referring here as the _systematic_ strategy of using indirect version strings through properties, even when they are used only once. 

Also, I realize that a benefit of indirect version strings is that then, all versions are listed at the same place, but this is, IMHO, compensated by a benefit of using direct version strings, i.e., that the direct ones are defined exactly at the place they are used, which is also a very good practice.

If you would not like to change this, would you perhaps consider commenting or referring to some post that justifies this practice (that, I find, is widely seen but seldom commented), for your user’s curiousity?

(In fact, there is another reason I ask, which is that I consider the possibility of proposing to merge my archetype with yours in order to join efforts, augment visibility and reduce dispersion by developing only one, as they have a common goal, but this is another story.)

**Comment by ngeor** [comment]
Hi Olivier,

I'm not really using/maintaing the archetype much these days. I think it's best indeed if you make the changes in your own archetype.

Regarding the rationale behind the properties, first I wanted to make sure the junit artifacts are on the same version and then later switched everything to properties for consistency. Otherwise some versions would be in properties and some would be direct, which might be messy. I've also thought that this might be a bit too bureaucratic for properties that are only used once, but I like the consistency (all versions are in properties) more. But I also don't have strong feelings about it to be honest.
